### PR TITLE
clean and enhance unit tests

### PR DESCRIPTION
**Pull request by jgrant22**
more cleansing. added test and made minor changes to the yaml

### CODE REVIEW

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 1]
```diff
@@ -204,12 +204,12 @@ public void allergy_identifier_test() {
...
 
     @Test
-    public void condition_PRB_identifier_test() {
+    public void conditionPrbIdentifierTest() {
         // Test with PV1-19.1 for visit number, no PRB-4
         String withoutPRB4 = "MSH|^~\\&|||||20040629164652|1|PPR^PC1|331|P|2.3.1||\n"
                 + "PID|||10290^^^WEST^MR||||20040530|M||||||||||||||||||||||N\n"
                 + "PV1||I||||||||SUR||||||||S|8846511^^^ACME|A|||||||||||||||||||SF|K||||20170215080000\n"
-                + "PRB|AD|2004062916460000|596.5^BLADDER DYSFUNCTION^I9||||20040629|||||||||20040629";
+                + "PRB|AD|20170110074000|K80.00^Cholelithiasis^I10||E1|1|20100907175347|20150907175347|20180310074000||||confirmed^Confirmed^http://terminology.hl7.org/CodeSystem/condition-ver-status|remission^Remission^http://terminology.hl7.org/CodeSystem/condition-clinical|20180310074000|20170102074000|textual representation of the time when the problem began|1^primary|ME^Medium|0.4|marginal|good|marginal|marginal|highly sensitive|some prb detail|\r";
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:212]: I took this segment from one of Joel's test. his changes made the original segment invalid.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 2]
```diff
@@ -224,14 +224,14 @@ public void condition_PRB_identifier_test() {
...
+        assertThat(type.getText()).isNull();
         assertThat(coding.getSystem()).isEqualTo("http://terminology.hl7.org/CodeSystem/v2-0203");
         assertThat(coding.getCode()).isEqualTo("VN");
         assertThat(coding.getDisplay()).isEqualTo("Visit number");
-
         // Test with no PV1-19, but with PID-18 for visit number; PRB-4.1 and PRB-4.3
         String withPRB4 = "MSH|^~\\&|||||20040629164652|1|PPR^PC1|331|P|2.3.1||\n" +
                 "PID|1||000054321^^^MRN||||19820512|M||2106-3|||||EN^English|M|CAT|78654||||N\n" +
-                "PRB|AD|2004062916460000|596.5^BLADDER DYSFUNCTION^I9|26744^^hithere|||20040629|||||||||20040629";
+                "PRB|AD|20170110074000|K80.00^Cholelithiasis^I10|53956|E1|1|20100907175347|20150907175347|20180310074000||||confirmed^Confirmed^http://terminology.hl7.org/CodeSystem/condition-ver-status|remission^Remission^http://terminology.hl7.org/CodeSystem/condition-clinical|20180310074000|20170102074000|textual representation of the time when the problem began|1^primary|ME^Medium|0.4|marginal|good|marginal|marginal|highly sensitive|some prb detail|\r";
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:234]: I took this segment from one of Joel's test. his changes made the original segment invalid.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 3]
```diff
@@ -246,55 +246,91 @@ public void condition_PRB_identifier_test() {
...
+        assertThat(type.getText()).isNull();
+        assertThat(coding.getSystem()).isEqualTo("http://terminology.hl7.org/CodeSystem/v2-0203");
+        assertThat(coding.getCode()).isEqualTo("VN");
+        assertThat(coding.getDisplay()).isEqualTo("Visit number");
+
+    }
+
+    @Test
+    public void conditionDg1IdentifierTest() {
+        // TODO: Add test for extId from DG1-3
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:258]: done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 4]
```diff
@@ -246,55 +246,91 @@ public void condition_PRB_identifier_test() {
...
+
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:259]: thanks! done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 5]
```diff
@@ -246,55 +246,91 @@ public void condition_PRB_identifier_test() {
...
+        assertThat(value).isEqualTo("C56.9-I10"); // DG1.3.1-DG1.3.3
         assertThat(system).isEqualTo("urn:id:extID");
 
-        // Test with MSH-7 for visit number, PRB-4.2
-        String msg = "MSH|^~\\&|||||20040629164652|1|PPR^PC1|331|P|2.3.1||\n"
-                + "PID|||10290^^^WEST\n"
-                + "PV1||I\n"
-                + "PRB|AD|2004062916460000|596.5^BLADDER DYSFUNCTION^I9|^kittykat|||20040629|||||||||20040629";
-        condition = ResourceUtils.getCondition(msg);
+        // Test with no PV1-19, but with PID-18 for visit number; PRB-4.1 and PRB-4.3
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:289]: thanks, done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 6]
```diff
@@ -246,55 +246,91 @@ public void condition_PRB_identifier_test() {
...
         assertThat(coding.getDisplay()).isEqualTo("Visit number");
 
-        // Identifier 2:  extID based on PRB-4.2
-        //        identifier = condition.getIdentifier().get(1);  // TODO, not getting second identifier
-        //        value = identifier.getValue();
-        //        system = identifier.getSystem();
-        //        assertThat(value).isEqualTo("kittykat"); // PRB-4.2
-        //        assertThat(system).isEqualTo("urn:id:extID");
-    }
+        identifier = condition.getIdentifier().get(1);
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:313]: added

[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:313]: added and noted

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 7]
```diff
@@ -246,55 +246,91 @@ public void condition_PRB_identifier_test() {
...
+        value = identifier.getValue();
+        system = identifier.getSystem();
+        assertThat(value).isEqualTo("B45678"); // DG1.3.1
+        assertThat(system).isEqualTo("urn:id:extID");
 
-    @Test
-    public void condition_DG1_identifier_test() {
-        // TODO: Add test for extId from DG1-3
+        identifier = condition.getIdentifier().get(2);
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:319]: added

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 8]
```diff
@@ -246,55 +246,91 @@ public void condition_PRB_identifier_test() {
...
+        value = identifier.getValue();
+        system = identifier.getSystem();
+        assertThat(value).isEqualTo("one"); // DG1.20.1
+        assertThat(system).isEqualTo("https://terminology.hl7.org/CodeSystem/two"); // DG1.20.2
+
+        identifier = condition.getIdentifier().get(3);
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:325]: added

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 9]
```diff
@@ -338,13 +374,12 @@ public void observation_identifier_test() {
...
 
         // Filler from ORC-3; OBX-3.2
-        msg = "MSH|^~\\&|SOARF|E_AMC|||201512201012||ADT^A01|MSG$1|P|2.7||1|||||||\r"
-                + "EVN|A01|201512201012||CI|wiestc|201512201007|1488\r"
-                + "PID|||MRN12345678^^^ID-XYZ^MR~111223333^^^USA^SS~MN1234567^^^MNDOT^DL||||||||||||||\r"
-                + "PV1|1|I|E2^E211^E211B^1488|Elective||||||||||V\r"
-                + "ORC|P789|F789\r"
-                + "OBR|1|ORD448811^NIST EHR|R-511^NIST Lab Filler|1000^Hepatitis A B C Panel^99USL|||20120628070100|||||||||\r"
-                + "OBX|1|ST|^ABC^||N/A||||||R||||||\r";
+        msg = "MSH|^~\\&|SendTest1|Sendfac1|Receiveapp1|Receivefac1|200603081747|security|PPR^PC1^PPR_PC1|1|P^I|2.6||||||ASCII||\r"
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:377]: I grabbed a PPR. message so ORC could be tested. I can simplify

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 10]
```diff
@@ -355,17 +390,16 @@ public void observation_identifier_test() {
...
-        // Placer from ORC-2; OBX-3.1
-        msg = "MSH|^~\\&|SOARF|E_AMC|||201512201012||ADT^A01|MSG$1|P|2.7||1|||||||\r"
-                + "EVN|A01|201512201012||CI|wiestc|201512201007|1488\r"
-                + "PID|||MRN12345678^^^ID-XYZ^MR~111223333^^^USA^SS~MN1234567^^^MNDOT^DL||||||||||||||\r"
-                + "PV1|1|I|E2^E211^E211B^1488|Elective||||||||||V\r"
-                + "ORC|P789\r"
-                + "OBR|1|ORD448811^NIST EHR|^NIST Lab Filler|1000^Hepatitis A B C Panel^99USL|||20120628070100|||||||||\r"
-                + "OBX|1|ST|HELLO^THERE||N/A||||||R||||||\r";
+        // Placer from ORC-2; OBX-3.2
+        msg = "MSH|^~\\&|SendTest1|Sendfac1|Receiveapp1|Receivefac1|200603081747|security|PPR^PC1^PPR_PC1|1|P^I|2.6||||||ASCII||\r"
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:397]: I grabbed a PPR. message so ORC could be tested. I can simplify

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 11]
```diff
@@ -355,17 +390,16 @@ public void observation_identifier_test() {
...
 
-        // Placer from ORC-2; OBX-3.1
-        msg = "MSH|^~\\&|SOARF|E_AMC|||201512201012||ADT^A01|MSG$1|P|2.7||1|||||||\r"
-                + "EVN|A01|201512201012||CI|wiestc|201512201007|1488\r"
-                + "PID|||MRN12345678^^^ID-XYZ^MR~111223333^^^USA^SS~MN1234567^^^MNDOT^DL||||||||||||||\r"
-                + "PV1|1|I|E2^E211^E211B^1488|Elective||||||||||V\r"
-                + "ORC|P789\r"
-                + "OBR|1|ORD448811^NIST EHR|^NIST Lab Filler|1000^Hepatitis A B C Panel^99USL|||20120628070100|||||||||\r"
-                + "OBX|1|ST|HELLO^THERE||N/A||||||R||||||\r";
+        // Placer from ORC-2; OBX-3.2
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:396]: okay

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 12]
```diff
@@ -733,63 +779,167 @@ public void procedure_identifier_test() {
...
+//        assertThat(system).isNull();
+//        type = identifier3.getType();
+//        coding = type.getCoding().get(0);
+//        assertThat(type.getText()).isNull();
+//        assertThat(coding.getSystem()).isEqualTo("http://terminology.hl7.org/CodeSystem/v2-0203");
+//        assertThat(coding.getCode()).isEqualTo("PLAC");
+//        assertThat(coding.getDisplay()).isEqualTo("Placer Identifier");
+//
+//          documentReference =
+//                  "MSH|^~\\&|HL7Soup|Instance1|MCM|Instance2|200911021022|Security|MDM^T02^MDM_T02|64322|P|2.6|123|456|ER|AL|USA|ASCII|en|2.6|56789^NID^UID|MCM||||\n" +
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:896]: done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 13]
```diff
@@ -733,63 +779,167 @@ public void procedure_identifier_test() {
...
+//        assertThat(type.getText()).isNull();
+//        assertThat(coding.getSystem()).isEqualTo("http://terminology.hl7.org/CodeSystem/v2-0203");
+//        assertThat(coding.getCode()).isEqualTo("FILL");
+//        assertThat(coding.getDisplay()).isEqualTo("Filler Identifier");
+//
+//        // Identifier 3: Placer
+//        Identifier identifier3 = report.getIdentifier().get(2);
+//        value = identifier3.getValue();
+//        system = identifier3.getSystem();
+//        assertThat(value).isEqualTo("PON001");
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:886]: done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 14]
```diff
@@ -733,63 +779,167 @@ public void procedure_identifier_test() {
...
+//        String value = identifier.getValue();
+//        String system = identifier.getSystem();
+//        assertThat(value).isEqualTo("200911021022");  // MSH-7
+//        assertThat(system).isEqualTo("urn:id:extID");
+//
+//        // Identifier 2: Filler
+//        identifier = report.getIdentifier().get(1);
+//        value = identifier.getValue();
+//        system = identifier.getSystem();
+//        assertThat(value).isEqualTo("FON001");
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:873]: added

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 15]
```diff
@@ -733,63 +779,167 @@ public void procedure_identifier_test() {
...
-    //    assertThat(system).isNull();
-    //    type = identifier3.getType();
-    //    coding = type.getCoding().get(0);
-    //    assertThat(coding.getSystem()).isEqualTo("http://terminology.hl7.org/CodeSystem/v2-0203");
-    //    assertThat(coding.getCode()).isEqualTo("PLAC");
-    //    assertThat(coding.getDisplay()).isEqualTo("Placer Identifier");
-    //
-    //  }
+//      @Test //Test works, but message type is not configured yet
+//      public void documentReferenceIdentifierTest() {
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:849]: added

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 16]
```diff
@@ -835,73 +987,125 @@ public void service_request_identifier_test1() {
...
 
         // Test 2: 
         //  - Visit number with PID-18
         //  - filler and placer from OBR
-        serviceRequest = "MSH|^~\\&|SendTest1|Sendfac1|Receiveapp1|Receivefac1|200603081747|security|OMP^O09^OMP_O09|1|P^I|2.6|||AL|NE|764|ASCII||||||^4086::132:2A57:3C28^IPv6\r"
-                + "PID|||12345^^^^MR||smith^john|||||||||||||112233\r"
-                + "ORC|NW\r"
-                + "OBR|1|2233|4455\r"
-                + "OBX|1|TX|^hunchback|1|Increasing||||||S\r";
+        serviceRequest = "MSH|^~\\&|SendTest1|Sendfac1|Receiveapp1|Receivefac1|200603081747|security|PPR^PC1^PPR_PC1|1|P^I|2.6||||||ASCII||\n" +
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:998]: the omp message does not have OBR segment switched to a PPR message, will simplify.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 17]
```diff
@@ -835,73 +987,125 @@ public void service_request_identifier_test1() {
...
         //  - Visit number with PV1-19
         //  - filler from OBR
         //  - placer from ORC
-        // TODO
+        serviceRequest = "MSH|^~\\&|SendTest1|Sendfac1|Receiveapp1|Receivefac1|200603081747|security|PPR^PC1^PPR_PC1|1|P^I|2.6||||||ASCII||\n" +
+                "PID|1||000054321^^^MRN|||||||||||||M|CAT|||||N\n" +
+                "PV1||I|6N^1234^A^GENERAL HOSPITAL2|||||||SUR||||||||S|8846511|A|||||||||||||||||||SF|K||||20170215080000\n" +
+                "PRB|AD|200603150625|aortic stenosis|53692||2||200603150625\n" +
+                "ORC||PON001||||E|^Q6H^D10^^^R\n" +
+                "OBR|1|CD150920001336|CD150920001336|||20150930000000|20150930164100|||||||||25055^MARCUSON^PATRICIA^L|||||||||F|||5755^DUNN^CHAD^B~25055^MARCUSON^PATRICIA^L|||WEAKNESS|DAS, SURJYA P||SHIELDS, SHARON A|||||||||";
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:1058]: done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 18]
```diff
@@ -835,73 +987,125 @@ public void service_request_identifier_test1() {
...
+        assertThat(coding.getDisplay()).isEqualTo("Placer Identifier");
 
         // Test 3: 
         //  - Visit number with PV1-19
         //  - filler from OBR
         //  - placer from ORC
-        // TODO
+        serviceRequest = "MSH|^~\\&|SendTest1|Sendfac1|Receiveapp1|Receivefac1|200603081747|security|PPR^PC1^PPR_PC1|1|P^I|2.6||||||ASCII||\n" +
+                "PID|1||000054321^^^MRN|||||||||||||M|CAT|||||N\n" +
+                "PV1||I|6N^1234^A^GENERAL HOSPITAL2|||||||SUR||||||||S|8846511|A|||||||||||||||||||SF|K||||20170215080000\n" +
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:1055]: will do thanks!

**Comment by klwhaley** [comment]
Merging since Jared is out unexpectedly and Lisa has thoroughly reviewed.

**Pull request merged**